src/checkbox.jsx

```diff
@@ -55,7 +55,12 @@ const Checkbox = React.createClass({
   //from the parent / owner using context
   componentWillReceiveProps(nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-    this.setState({muiTheme: newMuiTheme});
+    this.setState({
+      muiTheme: newMuiTheme,
+      switched: typeof nextProps.checked === 'boolean' && this.props.checked !== nextProps.checked
+        ? nextProps.checked
+        : this.state.switched,
+    });
```

Comment:
what if the component turns from being controlled to uncontrolled? It's very unlikely but if it does, people will have a hard time.